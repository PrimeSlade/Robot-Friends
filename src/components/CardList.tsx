import React from "react";
import Card from "./Card";
import { Robot } from "../type/Types";

type CardListProps = {
  robots: Array<Robot>;
};

const CardList = ({ robots }: CardListProps) => {
  return robots.map((user, i) => {
    return <Card key={i} id={user.id} name={user.name} email={user.email} />;
  });
};

export default CardList;
