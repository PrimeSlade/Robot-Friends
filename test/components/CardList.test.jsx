import { render } from "@testing-library/react";
import CardList from "../../src/components/CardList";

test("testing CardList by using snapshot", () => {
  // Mock robot data for testing
  const mockRobots = [
    {
      id: 1,
      name: "Leanne Graham",
      email: "incere@april.biz",
    },
    {
      id: 2,
      name: "Ervin Howell",
      email: "Shanna@melissa.tv",
    },
  ];
  const { asFragment } = render(<CardList robots={mockRobots} />);
  expect(asFragment()).toMatchSnapshot();
});
