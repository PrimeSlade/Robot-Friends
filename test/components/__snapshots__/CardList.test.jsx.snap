// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CardList by using snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robot"
      src="https://robohash.org/1?200*200"
    />
    <div>
      <h2>
        Leanne Graham
      </h2>
      <p>
        incere@april.biz
      </p>
    </div>
  </div>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robot"
      src="https://robohash.org/2?200*200"
    />
    <div>
      <h2>
        Ervin Howell
      </h2>
      <p>
        Shanna@melissa.tv
      </p>
    </div>
  </div>
</DocumentFragment>
`;
